File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character,Integer> map=new HashMap<>();
        for(char c:s.toCharArray()){
            map.put(c,map.getOrDefault(c,0)+1);
        }
        int ans=0;
        int found=0;
        for(Map.Entry<Character,Integer> m:map.entrySet()){
            ans+=m.getValue()/2 * 2;
            if(m.getValue()%2==1) found=1; 
        }
        if(found==1) ans+=1;
        return ans;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int> map;
        for(char c:s) map[c]++;
        int ans=0;
        int found=0;
        for(auto& m:map){
            ans+=m.second/2 * 2;
            if(m.second%2==1) found=1;
        }
        if(found) ans+=1;
        return ans;
    }
};